src/GlobalDiffEq.jl

```diff
@@ -25,6 +28,55 @@ function DiffEqBase.__solve(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase
     return sol
 end
 
-export GlobalRichardson
+struct GlobalAdjoint{A} <: GlobalDiffEqAlgorithm
+    alg::A
+end
+
+function DiffEqBase.__solve(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.AbstractDAEProblem},
+                            alg::GlobalAdjoint, args...;
+                            dt,kwargs...)
+
+    opt = Dict(kwargs)
+    otheropts = delete!(copy(opt), :dt)
+    tstops = get(opt, :tstops, range(prob.tspan[1], stop=prob.tspan[2], step=dt))
+    local sol
+
+    sol_init = solve(prob, alg.alg, args...; dt=dt, adaptive=false, otheropts...)
+    n = length(prob.u0)
+    C = 11/6 # for k = 2, 2 random vectors
+    E₂ = 2/π
+    Eₙ = vec_expt_full(n)
+    rtol = get(opt, :reltol, 1e-3)
+    atol = get(opt, :abstol, 1e-6)
+    gtol = atol # target gtol is abstol? abstol cannot be zero?
+    z₁, z₂ = orth_vec(n)
+
+    # g(x) = zᵀx
+    # g(u)= z' * u
+    # dg(u, t) = z' * du(t)
+    function dg₁(out,u,p,t,i) 
+        out .= z₁' * sol_init.k[i][1]
```

Comment:
is that the right loss?

Reply:
it's independent of the adjoint, so that doesn't seem right.

Reply:
So from the paper the adjoint equations should be: lm' = - J(u(t),t)^T*lm, lm(T) = z. I'm not sure how the setup used here would be equivalent to that. When I implemented this I just constructed that ODEProblem manually instead of trying to use adjoint sensitivity analysis. I think this is just confusion related to the remark near the algorithm outline at the end of the paper where they define a couple g functions that aren't related to the dg function in adjoint sensitivity analysis?

Reply:
Yes, you are correct. I thought that their g function was the loss function. I will try to take your approach using manual ODEProblems because I was struggling to set lm(T) = z using the DiffEqSensitivity package anyway.

Reply:
I am implementing a manual ODEProblem adjoint solution and will update this PR soon!